src/org/omegat/gui/exttrans/MachineTranslateTextArea.java

```diff
@@ -193,7 +195,12 @@ private String getTranslation(Language source, Language target) throws Exception
                     }
                 }
             }
-            return translator.getTranslation(source, target, src);
+            if (translator.isUsingGlossary()) {
+                List<GlossaryEntry> glossaryTerms = Core.getGlossaryManager().searchSourceMatches(currentlyProcessedEntry);
```

Comment:
Thanks, this is better wrt coupling.

This will recompute glossary terms for every provider that wants them. (Right now there’s only one, but that may change.) Instead it should only be computed once. 

But is it expensive? If not then I don’t see a need to muddy the provider API with `isUsingGlossary`: just compute it once and provide it to all; they can ignore if so desired.